fe/fe-core/src/main/java/com/starrocks/catalog/CatalogRecycleBin.java

```diff
@@ -974,6 +1039,79 @@ public void readFields(DataInput in) throws IOException {
         }
     }
 
+    public static class RecyclePartitionInfoV2 extends RecyclePartitionInfo  {
+        @SerializedName(value = "storageInfo")
+        protected StorageInfo storageInfo;
+
+        public RecyclePartitionInfoV2(long dbId, long tableId, Partition partition,
+                                      DataProperty dataProperty, short replicationNum, boolean isInMemory,
+                                      StorageInfo storageInfo) {
+            super(dbId, tableId, partition, dataProperty, replicationNum, isInMemory);
+            this.storageInfo = storageInfo;
+        }
+
+        public static RecyclePartitionInfoV2 read(DataInput in) throws IOException {
+            String json = Text.readString(in);
+            return GsonUtils.GSON.fromJson(json, RecyclePartitionInfoV2.class);
+        }
+
+        public StorageInfo getStorageInfo() {
+            return storageInfo;
+        }
+
+        @Override
+        public void write(DataOutput out) throws IOException {
+            out.writeLong(-1L);
+            String json = GsonUtils.GSON.toJson(this);
+            Text.writeString(out, json);
+        }
+    }
+
+    public static class RecycleRangePartitionInfo extends RecyclePartitionInfoV2 implements GsonPreProcessable,
```

Comment:
Need to add a new RuntimeTypeAdapterFactory in `GsonUtil` to indicate the inheritance relationship.